gen-models/src/operations.rs

```diff
@@ -233,11 +239,28 @@ pub struct OperationInfo {
     pub description: String,
 }
 
+pub fn calculate_file_checksum<P: AsRef<Path>>(file_path: P) -> Result<HashId, std::io::Error> {
+    let file = std::fs::File::open(file_path)?;
+    let reader = BufReader::new(file);
+    let hash_bytes = calculate_stream_hash(reader)?;
+    Ok(HashId(hash_bytes))
+}
+
+fn calculate_stream_hash<R: std::io::Read>(mut reader: R) -> Result<[u8; 32], std::io::Error> {
+    let mut hasher = Sha256::new();
+    io::copy(&mut reader, &mut hasher);
+    let result = hasher.finalize();
+    let mut hash_array = [0u8; 32];
+    hash_array.copy_from_slice(&result);
+    Ok(hash_array)
+}
+
 #[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
 pub struct FileAddition {
```

Comment:
we store a hash on both id and checksum. This is so empty files don't all have the same pointer, it's unique by (file_path, checksum)